plugins/UserCountry/API.php

```diff
@@ -89,33 +92,48 @@ public function getRegion($idSite, $period, $date, $segment = false)
     {
         $dataTable = $this->getDataTable(Archiver::REGION_RECORD_NAME, $idSite, $period, $date, $segment);
 
-        $segments = array('regionCode', 'countryCode');
-        $dataTable->filter('AddSegmentByLabel', array($segments, Archiver::LOCATION_SEPARATOR));
-
         $separator = Archiver::LOCATION_SEPARATOR;
         $unk = Visit::UNKNOWN_CODE;
 
-        // show visits tracked as Tibet as region of China
         $dataTables = [$dataTable];
 
         if ($dataTable instanceof DataTable\Map) {
             $dataTables = $dataTable->getDataTables();
         }
 
         foreach ($dataTables as $dt) {
-            if ($dt->getRowFromLabel('1|ti')) {
+            $archiveDate = $dt->getMetadata(DataTable::ARCHIVED_DATE_METADATA_NAME);
+
+            // convert fips region codes to iso if required
+            if ($this->shouldRegionCodesBeConvertedToIso($archiveDate, $date, $period)) {
                 $dt->filter('GroupBy', array(
                     'label',
-                    function ($label) {
-                        if ($label == '1|ti') {
-                            return '14|cn';
```

Comment:
I guess this hard-coded conversion here and start from line 172 is still required for the old format.

Reply:
True. My original implementation converted all codes so it wasn't needed anymore, but with this implementation it's still needed. Will add it again